### DISCUSSION TITLE
the beginning of a call is cut off

### DISCUSSION THREAD

**Discussion by tomo90** [juha-h/baresip-studio#352] [Q&A]
Hi,

my VOIP provider plays information about the number of minutes remaining on my account before every outgoing call. For some reason, Baresip cuts off the beginning of this message. The message says e.g. "You have eighteen minutes remaining." When I call using Baresip I only hear something like "...teen minutes remaining". 

None of the SIP clients I have tried so far did this. I have tried Linphone (Windows, Android), Jami (Windows, Android), MicroSIP (Windows), Sipnetic (Android). 

Any ideas? :)

**Comment by juha-h** [comment]
The issue is this: https://stackoverflow.com/questions/47676936/audiomanager-is-introducing-delay-in-the-setmodemode-in-communication and I also created a Google issue  about it: https://issuetracker.google.com/issues/133867453.

Which Android version you have in your device?  If 12 or greater, baresip can detect when audio mode changes to communication.  Below that I could add some delay before actually making the call and hope that the change has been done.

**Reply by tomo90** [reply]
I'm using Android 13 (LineageOS 20) on an older device - Xiaomi Redmi 3s (from 2016).

**Comment by juha-h** [comment]
It should work OK with Android 13.  Try to call thetestcall@opensips.org.  I just did with both baresip and Linphone, and I heard "One moment please ..." with both apps.

**Comment by tomo90** [comment]
OK, tried that
With Baresip I heard just "..oment please..."
With Linphone I heard the whole message..

**Comment by juha-h** [comment]
May be in case of your device some amount of delay needs to be added also when Android version is 12+.  But it would get too messy.  Looks like baresip is not suitable for your device.

**Comment by juha-h** [comment]
I created a test app https://box.tutpro.com/tmp/audiodelay.apk.  It should show three play buttons.  First plays digits right after audio mode is changed from normal to communication and second waits after the change has taken place.  Third plays them in normal mode.  Do you hear all digits with all buttons?

**Comment by tomo90** [comment]
PLAY IN_COMMUNICATION - I hear "..7 8 9 10"
PLAY IN_COMMUNICATION AFTER CHANGE - I hear "..6 7 8 9 10"
PLAY NORMAL - I hear "1 2 3 4 5 6 7 8 9 10"

**Comment by juha-h** [comment]
On my Samsung A04 device, I hear all digits with all three buttons.  I created new version of [https://box.tutpro.com/tmp/audiodelay.apk](https://box.tutpro.com/tmp/audiodelay.apk) that has button PLAY IN_COMMUNICATION_AFTER_DELAY.  Try with that to find out how big the delay needs to be so that you hear all digits.

**Comment by tomo90** [comment]
Unfortunately the app crashes on touching the PLAY IN_COMMUNICATION_AFTER_DELAY button.

**Comment by juha-h** [comment]
tomo90 writes:

> Unfortunately the app crashes on touching the PLAY
> IN_COMMUNICATION_AFTER_DELAY button.

Did you give a delay value, such as 1000?

**Comment by tomo90** [comment]
Right, that's how it works, of course..sorry, my bad. 
The lowest threshold where I can hear all the numbers is 1800 ms.

**Comment by juha-h** [comment]
Thanks for testing.  Since some devices don't need any delay, it would not be a good idea to force the delay to all devices.  Perhaps a new setting would be needed to solve the issue.

**Reply by tomo90** [reply]
No problem, i get it. 
BTW, you are doing fantastic job with baresip 👍

**Comment by juha-h** [comment]
@tomo90 I added in Settings->Audio Settings a new setting `Audio Delay` with default value 100 (ms).  Could you try if the settings solves this issue on your device: https://box.tutpro.com/tmp/baresip.apk.  If yes, with what value?

**Comment by tomo90** [comment]
Seems good! The problem is solved by setting the value to 500 ms.

**Comment by juha-h** [comment]
tomo90 writes:

> Seems good! The problem is solved by setting the value to 500 ms.

Thanks for testing.  I'll use 500 as default value if Android version is
new enough to detect audio mode changes and 1500 otherwise.

**Comment by juha-h** [comment]
Remedy is now available in version 54.1.0.
